crates/transpiler/src/target/mod.rs

```diff
@@ -1396,43 +1365,72 @@ impl Target {
-                                    .all(|qarg| qarg.0 < self.num_qubits.unwrap_or_default());
-                        }
+
+                let obj_params = obj.params();
+                if parameters.len() != obj_params.len() {
+                    return false;
+                }
+
+                for (index, params) in parameters.iter().enumerate() {
+                    let obj_at_index = &obj_params[index];
+                    let matching_params = match (obj_at_index, params) {
+                        (Param::Float(obj_f), Param::Float(param_f)) => obj_f == param_f,
+                        (Param::ParameterExpression(_), _) => true,
+                        _ => Python::attach(|py| python_compare(py, params, &obj_params[index]))
+                            .expect("Error comparing Python parameters."),
+                    };
+
+                    if !matching_params {
+                        return false;
                     }
                 }
-            } else {
-                // Duplicate case is if it contains none
-                let obj = &self._gate_name_map[operation_name];
+                if check_angle_bounds
+                    && self.has_angle_bounds()
+                    && parameters.iter().all(|x| matches!(x, Param::Float(_)))
+                {
+                    let params: Vec<f64> = parameters
+                        .iter()
+                        .map(|x| {
+                            let Param::Float(val) = x else { unreachable!() };
+                            *val
+                        })
+                        .collect();
+                    if self.angle_bounds.contains_key(operation_name)
+                        && !self.gate_supported_angle_bound(operation_name, &params)
+                    {
+                        return false;
+                    }
+                }
+            }
+            let QargsRef::Concrete(qargs_as_vec) = qargs else {
+                return true;
+            };
+            if self.gate_map[operation_name].contains_key(&qargs) {
+                return true;
+            }
+            if self.gate_map.get(operation_name).is_none()
+                || self.gate_map[operation_name].contains_key(&QargsRef::Global)
+            {
                 match obj {
                     TargetOperation::Variadic(_) => {
-                        return qargs.is_global()
-                            || qargs_as_vec
-                                .iter()
-                                .all(|qarg| qarg.0 <= self.num_qubits.unwrap_or_default())
-                                && qarg_set.len() == qargs_as_vec.len();
+                        let qarg_set: HashSet<&PhysicalQubit> = qargs_as_vec.iter().collect();
+                        return qargs_as_vec
+                            .iter()
+                            .all(|qarg| qarg.0 <= self.num_qubits.unwrap_or_default())
+                            && qarg_set.len() == qargs_as_vec.len();
```

Comment:
Why do we need to check for duplicates?

Reply:
I only did it because we did this initially but I didn't give it much thought. I removed it and everything seems to work just fine.